core/src/main/scala/io/qbeast/spark/index/OTreeDataAnalyzer.scala

```diff
@@ -225,6 +227,28 @@ object DoublePassOTreeDataAnalyzer
     }
     logDebug(s"revisionToUse=$revisionToUse")
 
+    // Check if the DataFrame and the Columns To Index are deterministic
```

Comment:
Create a method for `DoublePassOTreeDataAnalyzer` that handles all these checks.

Reply:
sure